lib/nmatrix/shortcuts.rb

```diff
@@ -236,6 +236,86 @@ def eye(shape, opts={})
       m
     end
     alias :identity :eye
+
+    #
+    # call-seq:
+    #     hilbert(shape) -> NMatrix
+    #     hilbert(shape, dtype: dtype) -> NMatrix
+    #     hilbert(shape, stype: stype, dtype: dtype) -> NMatrix
+    #
+    # Creates an hilbert matrix (square matrix).
+    #
+    # * *Arguments* :
+    #   - +size+ -> integer ( for square matrix) specifying the dimensions.
+    #   - +dtype+ -> (optional) Default is +:float64+
+    #   - +stype+ -> (optional) Default is +:dense+.
+    # * *Returns* :
+    #   - A hilbert matrix.
+    #
+    # Examples:
+    #
+    #    NMatrix.hilbert(3) # =>  1.0     0.5      0.3333333333333333
+    #            0.5                         0.3333333333333333    0.25
+    #            0.3333333333333333          0.25                  0.2
+    #
+    def hilbert(shape, opts={})
+      # Fill the diagonal with 1's.
+      m = NMatrix.ones(shape, {:dtype => :float64}.merge(opts))
+      (0...(shape)).each do |i|
+          (0...(i+1)).each do |j|
+            m[i,j]=m[i,j]/(j+i+1)
+            if i!=j
+              m[j,i]=m[i,j]
+            end
+          end
+       end
+      m
+    end
+
+    #
+    # call-seq:
+    #     inv_hilbert(shape) -> NMatrix
+    #     inv_hilbert(shape, dtype: dtype) -> NMatrix
+    #     inv_hilbert(shape, stype: stype, dtype: dtype) -> NMatrix
+    #
+    # Creates an inverse hilbert matrix (square matrix rank 2).
+    #
+    # * *Arguments* :
+    #   - +size+ -> Array (or integer for square matrix) specifying the dimensions.
+    #   - +dtype+ -> (optional) Default is +:float64+
+    #   - +stype+ -> (optional) Default is +:dense+.
+    # * *Returns* :
+    #   - A hilbert matrix.
+    #
+    # Examples:
+    #    NMatrix.inv_hilbert(3) # =>   9.0,  -36.0,   30.0
+    #                          -36.0,  192.0, -180.0
+    #                          30.0, -180.0,  180.0
+    #
+    #
+    def inv_hilbert(shape, opts={})
+      # Fill the diagonal with 1's.
+      m = NMatrix.ones(shape, {:dtype => :float64}.merge(opts))
+      comb = NMatrix.ones(shape*2, {:dtype => :float64}.merge(opts))
+      (0...(shape*2)).each do |i|
+          (0...(i+1)).each do |j|
+            if j!=0 and j !=i
+              comb[i,j]=comb[i-1,j]+comb[i-1,j-1]
+            end
+          end
+       end
+
+      (0...shape).each do |i|
+          (0...i+1).each do |j|
+            m[i,j]=(-1)**((i+j))*((i+j)+1)*comb[shape + i,shape - j - 1]*comb[shape + j,shape - i - 1]*comb[(i+j),i]*comb[(i+j),i]
```

Comment:
This expression is huge, so could you enclose the larger parts inside parentheses and break it up into two or more lines?

It affects readability adversely.